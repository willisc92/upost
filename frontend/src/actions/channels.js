import API from "../utils/API";

// SET_CHANNELS
export const setChannels = (channels) => ({
    type: "SET_CHANNELS",
    channels
});

// START_SET_CHANNELS
export const startSetChannels = () => {
    return (dispatch) => {
        API.get("channels/", {
            params: {
                user: localStorage.getItem("user_id")
            }
        }).then((result) => {
            dispatch(setChannels(result.data));
        });
    };
};

export const addChannel = (channel) => {
    return (dispatch) => {
        API.post("channels/", {
            ...channel,
            user: localStorage.getItem("user_id")
        })
            .then((result) => {
                dispatch(startSetChannels());
            })
            .catch((err) => {
                alert(err);
            });
    };
};
