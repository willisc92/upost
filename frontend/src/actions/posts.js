import API, { setContentToForm, resetContentType } from "../utils/API";
import { getCurrentUser } from "./auth";
import moment from "moment";

/**
 * POST_START.
 * action generator.
 * Resets the error and sets loading status.
 *
 * @returns {Object} object to reset error and loading status
 */
export const postStart = () => ({
    type: "POST_START"
});

/**
 * SET_POSTS.
 * action generator.
 * adds user's posts to store.
 *
 * @param {Object[]} posts posts to add to store
 * @returns {Object} object to add posts to store
 */
export const setPosts = (posts) => ({
    type: "SET_POSTS",
    posts
});

/**
 * START_SET_POSTS.
 * from API gets the current user's posts and adds to store.
 *
 * @returns dispatch function
 */
export const startSetMyPosts = () => {
    return (dispatch) => {
        dispatch(postStart());
        getCurrentUser()
            .then((res) => {
                API.get("posts/", {
                    params: {
                        user: res.data.id
                    }
                })
                    .then((result) => {
                        dispatch(postSuccess());
                        dispatch(setPosts(result.data));
                    })
                    .catch((err) => {
                        dispatch(postFail(err));
                    });
            })
            .catch((err) => {
                console.log(JSON.stringify(err));
            });
    };
};

/**
 * SET_INTEREST_RANDOM_POSTS.
 * action generator.
 * resets the interestRandomPosts in store to be an empty array.
 *
 * @returns {Object} object to reset interestRandomPosts
 */
export const setInterestRandomPosts = () => {
    return { type: "SET_INTEREST_RANDOM_POSTS" };
};

/**
 * ADD_INTEREST_RANDOM_POSTS.
 * action generator.
 * adds a new interest and associated random posts to store.
 *
 * @param {Object} interestPosts contains tag and an array of posts
 * @returns {Object} object to add interest and posts to interestRandomPosts
 */
export const addInterestRandomPosts = (interestPosts) => {
    return { type: "ADD_INTEREST_RANDOM_POSTS", interestPosts };
};

/**
 * START_SET_INTEREST_RANDOM_POSTS.
 * resets the interestRandomPosts store.
 * for each interest provided gets random posts from API and adds to store.
 *
 * @param {Object[]} interests an array of interest objects or interest tags
 * @returns {Function} dispatch function
 */
export const startSetInterestRandomPosts = (interests) => {
    return (dispatch) => {
        return new Promise((resolve, reject) => {
            dispatch(postStart());
            dispatch(setInterestRandomPosts());
            // if we have a list of interest objects
            if (interests.length > 0 && !!interests[0].interest_tag) {
                for (let i = 0; i < interests.length; i++) {
                    API.get("random-posts/", {
                        params: {
                            interest: interests[i].interest_tag
                        }
                    }).then((result) => {
                        if (result.data.length > 0) {
                            dispatch(addInterestRandomPosts({ tag: interests[i].interest_tag, posts: result.data }));
                        }
                    });
                }
            }
            // if we have a list of interest tags
            else {
                for (let i = 0; i < interests.length; i++) {
                    API.get("random-posts/", {
                        params: {
                            interest: interests[i]
                        }
                    }).then((result) => {
                        if (result.data.length > 0) {
                            dispatch(addInterestRandomPosts({ tag: interests[i], posts: result.data }));
                        }
                    });
                }
            }
            dispatch(postSuccess());
            resolve(true);
        });
    };
};

/**
 * START_GET_POST.
 * from API gets a specifc post and adds to store.
 *
 * @param {number} id post id to get
 * @returns {Promise}
 */
export const startGetPost = (id) => {
    return (dispatch) => {
        return new Promise((resolve, reject) => {
            dispatch(postStart());
            API.get("posts/", {
                params: {
                    post_id: id
                }
            })
                .then((result) => {
                    dispatch(postSuccess());
                    dispatch(setPosts(result.data));
                    resolve(result);
                })
                .catch((err) => {
                    dispatch(postFail(err));
                    reject(err);
                });
        });
    };
};

/**
 * GET_POST.
 * from API gets a specific post, but does not add to store.  Does not dispatch any actions.
 *
 * @param {number} id post id to get
 */
export const getPost = (id) => {
    return new Promise((resolve, reject) => {
        API.get("posts/", {
            params: {
                post_id: id
            }
        })
            .then((result) => {
                resolve(result.data);
            })
            .catch((err) => {
                reject(err);
            });
    });
};

/**
 * ADD_POST.
 * sends post request to API to add a new user created post.
 * upon sucessful post reloads user posts.
 *
 * @param {Object} post post object to add to API
 * @returns {Promise}
 */
export const addPost = (post) => {
    return (dispatch) => {
        return new Promise((resolve, reject) => {
            if (post.toString() == "[object FormData]") {
                setContentToForm();
            }
            dispatch(postStart());
            API.post(`posts/`, post)
                .then((result) => {
                    if (post.toString() == "[object FormData]") {
                        resetContentType();
                    }
                    dispatch(postSuccess());
                    resolve(result);
                })
                .catch((err) => {
                    resetContentType();
                    dispatch(postFail(err));
                    reject(err);
                });
        });
    };
};

/**
 * EDIT_POST.
 * sends put request to API to edit a user's post.
 * upon sucessful post reloads user posts.
 *
 * @param {number} id post id to edit
 * @param {Object} updates updates to be made to the post
 */
export const editPost = (id, updates) => {
    return (dispatch) => {
        return new Promise((resolve, reject) => {
            if (updates.toString() == "[object FormData]") {
                setContentToForm();
            }
            dispatch(postStart());
            API.patch(`posts/${id}/`, updates)
                .then((result) => {
                    if (updates.toString() == "[object FormData]") {
                        resetContentType();
                    }
                    dispatch(postSuccess());
                    resolve(result);
                })
                .catch((err) => {
                    resetContentType();
                    dispatch(postFail(err));
                    reject(err);
                });
        });
    };
};

/**
 * POST_FAIL.
 * action generator.
 * adds error to the store.
 *
 * @param {Object} error the error
 * @returns {Object} object to set error
 */
export const postFail = (error) => ({
    type: "POST_FAIL",
    error
});

/**
 * POST_SUCCESS.
 * action generator.
 * resets loading and error status.
 *
 * @returns {Object} object to reset loading and error status
 */
export const postSuccess = () => ({
    type: "POST_SUCCESS"
});

/**
 * CLEAR_POSTS.
 * action generator.
 * clears posts in store
 *
 * @returns {Object} object to reset posts
 */
export const clearPosts = () => ({
    type: "CLEAR_POSTS"
});

/**
 * searchPosts
 * action dispatcher - searches for posts
 * @param {string} text - text to search posts by
 * @returns {Promise} to be handled
 */
export const searchPosts = (text) => {
    return (dispatch) => {
        return new Promise((resolve, reject) => {
            dispatch(postStart());
            API.get(`posts/?search=${text}`)
                .then((result) => {
                    dispatch(postSuccess());
                    dispatch(setPosts(result.data));
                    resolve(result);
                })
                .catch((err) => {
                    dispatch(postFail(err));
                    reject(err);
                });
        });
    };
};

/**
 * deletePost
 * action dispatcher - soft deletes a post
 * @param {number} id - id of post to delete
 * @param {Object} post - Post object to delete
 * @returns {Promise} to be handled
 */
export const deletePost = (id) => {
    return (dispatch) => {
        return new Promise((resolve, reject) => {
            dispatch(postStart());
            const updates = {
                deleted_flag: true,
                deletion_date: moment().toDate()
            };
            API.patch(`posts/${id}/`, updates)
                .then((result) => {
                    dispatch(postSuccess());
                    resolve(result);
                })
                .catch((err) => {
                    dispatch(postFail(err));
                    reject(err);
                });
        });
    };
};

/**
 * restorePost
 * action dispatcher - restores a post
 * @param {number} id - id of post to restore
 * @param {Object} post - Post object to restore
 * @returns {Promise} to be handled
 */
export const restorePost = (id) => {
    return (dispatch) => {
        return new Promise((resolve, reject) => {
            dispatch(postStart());
            const updates = {
                deleted_flag: false,
                deletion_date: null
            };
            API.patch(`posts/${id}/`, updates)
                .then((result) => {
                    dispatch(postSuccess());
                    resolve(result);
                })
                .catch((err) => {
                    dispatch(postFail(err));
                    reject(err);
                });
        });
    };
};

/**
 * getNonInterestPosts
 * action dispatcher - gets posts that are NOT relavent to your interests
 * @returns {Promise} to be handled
 */
export const getNonInterestPosts = () => {
    return (dispatch) => {
        return new Promise((resolve, reject) => {
            dispatch(postStart());
            API.get(`non-interest-posts/`)
                .then((result) => {
                    dispatch(postSuccess());
                    dispatch(setNonInterestPosts(result.data));
                    resolve(result);
                })
                .catch((err) => {
                    dispatch(postFail(err));

                    reject(err);
                });
        });
    };
};

/**
 * setNonInterestPosts
 * action generator - sets posts not relavent to interests
 * @returns {Object} Action object
 */
export const setNonInterestPosts = (nonInterestPosts) => ({
    type: "SET_NON_INTEREST_POSTS",
    nonInterestPosts
});

/**
 * getRandomNonInterestPost
 * action dispatcher - Gets a random post that is not relavent to interests
 * @returns {Promise} Promise to handle
 */
export const getRandomNonInterestPost = () => {
    return (dispatch) => {
        return new Promise((resolve, reject) => {
            dispatch(postStart());
            API.get(`random-post/`)
                .then((result) => {
                    dispatch(postSuccess());
                    resolve(result);
                })
                .catch((err) => {
                    dispatch(postFail(err));
                    reject(err);
                });
        });
    };
};

/**
 * getCommunityPosts
 * action dispatcher - Gets posts only according to interests
 * @returns {Promise} Promise to handle
 */
export const getCommunityPosts = () => {
    return (dispatch) => {
        return new Promise((resolve, reject) => {
            dispatch(postStart());
            API.get(`community-posts/`)
                .then((result) => {
                    dispatch(postSuccess());
                    dispatch(setPosts(result.data));
                    resolve(result);
                })
                .catch((err) => {
                    dispatch(postFail(err));
                    reject(err);
                });
        });
    };
};
