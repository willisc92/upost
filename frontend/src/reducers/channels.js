// Channels Reducers
const channelsDefaultState = [];

export default (state = channelsDefaultState, action) => {
    switch (action.type) {
        case "SET_CHANNELS":
            return {
                channels: action.channels
            };
        default:
            return state;
    }
};
