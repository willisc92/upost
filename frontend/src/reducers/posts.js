// Post Reducers
export const postsDefaultState = { error: null, loading: false, posts: [], interestRandomPosts: [] };

export default (state = postsDefaultState, action) => {
    switch (action.type) {
        case "POST_START":
            return {
                ...state,
                error: null,
                loading: true
            };
        case "SET_POSTS":
            return {
                ...state,
                posts: action.posts
            };
        case "POST_FAIL":
            return {
                ...state,
                error: action.error,
                loading: false
            };
        case "POST_SUCCESS":
            return {
                ...state,
                error: null,
                loading: false
            };
        case "SET_INTEREST_RANDOM_POSTS":
            return {
                ...state,
                interestRandomPosts: []
            };
        case "ADD_INTEREST_RANDOM_POSTS":
            return {
                ...state,
                interestRandomPosts: [...state.interestRandomPosts, action.interestPosts]
            };
        default:
            return state;
    }
};
