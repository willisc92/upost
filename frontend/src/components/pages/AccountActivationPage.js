import React from "react";
import { startActivation } from "../../actions/activate";

class AccountActivationPage extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            inProgress: true,
            isValid: undefined
        };
    }

    componentDidMount() {
        const uid = this.props.match.params.uid;
        const token = this.props.match.params.token;
        startActivation(uid, token)
            .then((result) => {
                this.setState(() => {
                    return { inProgress: false, isValid: true };
                });
            })
            .catch((error) => {
                console.log("error", JSON.stringify(error, null, 2));
                this.setState(() => {
                    return { inProgress: false, isValid: false };
                });
            });
    }

    render() {
        return (
            <div>
                <div className="page-header">
                    <div className="content-container">
                        {this.state.inProgress ? (
                            <h1 className="page-header__title">
                                Your <span>UPost</span> account is currently being activated!
                            </h1>
                        ) : this.state.isValid ? (
                            <h1 className="page-header__title">
                                Thank you for your email confirmation. Please login and complete your registration.
                            </h1>
                        ) : (
                            <h1 className="page-header__title">The activation link is invalid</h1>
                        )}
                    </div>
                </div>
            </div>
        );
    }
}

export default AccountActivationPage;
