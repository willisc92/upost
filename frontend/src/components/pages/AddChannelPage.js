import React from "react";
import ChannelForm from "../forms/ChannelForm";
import { connect } from "react-redux";
import { addChannel } from "../../actions/channels";

class AddChannelPage extends React.Component {
    constructor(props) {
        super(props);
    }

    onSubmit = (channel) => {
        this.props
            .addChannel({ ...channel })
            .then(() => this.props.history.push("/myChannels"))
            .catch(() => {});
    };

    render() {
        return (
            <div>
                <div className="page-header">
                    <div className="content-container">
                        <h1 className="page-header__title">Add a Channel</h1>
                    </div>
                </div>
                <div className="content-container">
                    <ChannelForm onSubmit={this.onSubmit} />
                </div>
            </div>
        );
    }
}

const mapDispatchToProps = (dispatch) => ({
    addChannel: (channel) => dispatch(addChannel(channel))
});

export default connect(
    null,
    mapDispatchToProps
)(AddChannelPage);
