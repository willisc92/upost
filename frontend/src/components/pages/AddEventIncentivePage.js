import React from "react";
import { addIncentivePackage } from "../../actions/incentivePackage";
import { getCurrentUser } from "../../actions/auth";
import IncentiveForm from "../forms/IncentiveForm";
import { connect } from "react-redux";
import { startSetEvent, clearEvents } from "../../actions/events";

import Box from "@material-ui/core/Box";
import Container from "@material-ui/core/Container";
import Typography from "@material-ui/core/Typography";
import Button from "@material-ui/core/Button";

class AddEventIncentivePage extends React.Component {
    constructor(props) {
        super(props);
    }

    componentWillMount() {
        this.props.clearEvents();
        const event_id = this.props.match.params.id;
        getCurrentUser()
            .then((res) => {
                this.props
                    .startSetEvent(event_id)
                    .then((event_res) => {
                        if (res.data.username !== event_res.data.event_owner) {
                            this.props.history.push(`/myChannels`);
                        }
                    })
                    .catch((err) => {
                        console.log(JSON.stringify(err, null, 2));
                    });
            })
            .catch((err) => {
                console.log(JSON.stringify(err, null, 2));
            });
    }

    onSubmit = (incentive) => {
        const post_id = this.props.event.post;
        this.props
            .addIncentivePackage(incentive)
            .then((res) => this.props.history.push(`/post-events/${post_id}`))
            .catch((err) => {
                console.log(err);
            });
    };

    goBack = () => {
        const event_id = this.props.match.params.id;
        const post_id = this.props.event.post;
        this.props.history.push(`/myPosts/${post_id}/events/${event_id}/edit`);
    };

    goToIncentive = () => {
        const event_id = this.props.match.params.id;
        this.props.history.push(`/myEvents/${event_id}/editIncentive`);
    };

    render() {
        const read_only = !!this.props.event && this.props.event.deleted_flag;
        const existing_incentive = !!this.props.event && !!this.props.event.event_incentive;

        return (
            !!this.props.event && (
                <div>
                    <Box bgcolor="secondary.main" py={3}>
                        <Container fixed>
                            <Typography gutterBottom variant="h2">
                                Add an Incentive Package to Event:{" "}
                                <Typography variant="inherit" display="inline" color="primary">
                                    {this.props.event && this.props.event.event_title}
                                </Typography>
                            </Typography>

                            {existing_incentive ? (
                                <Box paddingBottom={2}>
                                    <Typography variant="h2" color="error" gutterBottom>
                                        There is already an existing incentive tied to this event.
                                    </Typography>
                                    <Button variant="contained" color="primary" onClick={this.goToIncentive}>
                                        Go to Incentive
                                    </Button>
                                </Box>
                            ) : (
                                read_only && (
                                    <Typography variant="h2" color="error" gutterBottom>
                                        The event this incentive will be tied to is deleted. Restore it before adding an
                                        incentive.
                                    </Typography>
                                )
                            )}
                            <Box>
                                <Button variant="contained" color="primary" onClick={this.goBack}>
                                    Go to Event
                                </Button>
                            </Box>
                        </Container>
                    </Box>
                    <Container fixed>
                        <IncentiveForm
                            onSubmit={this.onSubmit}
                            nextStep="Save"
                            event={this.props.event}
                            fromEvent={true}
                            read_only={read_only || existing_incentive}
                        />
                    </Container>
                </div>
            )
        );
    }
}

const mapStateToProps = (state) => ({
    event: state.events.events.length !== 0 ? state.events.events : null
});

const mapDispatchToProps = (dispatch) => ({
    clearEvents: () => dispatch(clearEvents()),
    addIncentivePackage: (incentive) => dispatch(addIncentivePackage(incentive)),
    startSetEvent: (id) => dispatch(startSetEvent(id))
});

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(AddEventIncentivePage);
