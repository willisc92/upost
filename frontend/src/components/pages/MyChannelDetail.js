import React from "react";
import { startGetChannel, deleteChannel, restoreChannel } from "../../actions/channels";
import { deletePost } from "../../actions/posts";
import { connect } from "react-redux";
import moment from "moment";
import MyPostFilterSelector from "../filter_selectors/PostFilterSelector";
import { getVisiblePosts } from "../../selectors/myPosts";
import { getCurrentUser } from "../../actions/auth";
import MyPostSummary from "../MyPostSummary";
import Box from "@material-ui/core/Box";
import Button from "@material-ui/core/Button";
import Typography from "@material-ui/core/Typography";
import Container from "@material-ui/core/Container";

export class MyChannelDetail extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            selected: 0
        };
    }

    componentDidMount() {
        const channel_id = this.props.match.params.id;
        getCurrentUser()
            .then((res) => {
                this.props
                    .startGetChannel(channel_id)
                    .then((channel_res) => {
                        if (res.data.username !== channel_res.data[0].user) {
                            this.props.history.push("/myChannels");
                        }
                    })
                    .catch((err) => {
                        console.log(JSON.stringify(err, null, 2));
                    });
            })
            .catch((err) => {
                console.log(JSON.stringify(err, null, 2));
            });
    }

    onSelect = (key) => {
        this.setState({ selected: key });
    };

    handleAddPost = () => {
        const channel_id = this.props.match.params.id;
        this.props.history.push(`/myChannels/${channel_id}/addPost`);
    };

    handleEditChannel = (e) => {
        const channel_id = this.props.match.params.id;
        this.props.history.push(`/myChannels/edit/${channel_id}`);
    };

    restoreChannel = (e) => {
        const channel_id = this.props.match.params.id;
        this.props
            .restoreChannel(channel_id)
            .then(() => {
                this.props.history.push(`/myChannels/${channel_id}/`);
            })
            .catch((err) => {
                console.log(JSON.stringify(err, null, 2));
            });
    };

    deleteChannel = (e) => {
        const channel_id = this.props.match.params.id;
        this.props
            .deleteChannel(channel_id)
            .then(() => {
                this.props.history.push(`/myChannels/${channel_id}/`);
            })
            .catch((err) => {
                console.log(JSON.stringify(err, null, 2));
            });
    };

    deleteAllPosts = (e) => {
        let promises = [];
        const posts = this.props.posts;

        posts.forEach((post) => {
            promises.push(this.props.deletePost(post.post_id));
        });

        const channel_id = this.props.match.params.id;

        Promise.all(promises)
            .then(() => {
                this.props.startGetChannel(channel_id);
            })
            .catch((err) => {
                console.log(err);
                console.log(JSON.stringify(err, null, 2));
            });
    };

    render() {
        const posts = !!this.props.posts && this.props.posts;

        return (
            <div>
                <Box bgcolor="secondary.main" py={3}>
                    <Container fixed>
                        <Box paddingBottom={2}>
                            <Typography variant="h1" display="inline" gutterBottom>
                                Channel Page:{" "}
                            </Typography>
                            <Typography variant="h1" display="inline" color="primary" gutterBottom>
                                {this.props.channel.channel_name}
                            </Typography>
                        </Box>
                        {!!this.props.channel.channel_description && (
                            <Typography variant="h2" gutterBottom>
                                Description: {this.props.channel.channel_description}
                            </Typography>
                        )}
                        <Typography variant="h3" gutterBottom>
                            Creation Date: {moment(this.props.channel.creation_date).format("MMMM Do YYYY")}
                        </Typography>
                        {this.props.channel.deleted_flag && (
                            <Typography variant="h3" color="error" gutterBottom>
                                Deletion Date: {moment(this.props.channel.deletion_date).format("MMMM Do YYYY")} -
                                Restore the Channel To Add/Edit Posts
                            </Typography>
                        )}

                        {this.props.channel.deleted_flag ? (
                            <Button color="primary" variant="contained" onClick={this.restoreChannel}>
                                Restore Channel
                            </Button>
                        ) : (
                            <div>
                                <Button color="primary" variant="contained" onClick={this.handleAddPost}>
                                    Add a new post
                                </Button>{" "}
                                <Button
                                    id={this.props.channel.channel_id}
                                    color="primary"
                                    variant="contained"
                                    onClick={this.handleEditChannel}
                                >
                                    Edit this channel
                                </Button>{" "}
                                <Button
                                    id={this.props.channel.channel_id}
                                    color="primary"
                                    variant="contained"
                                    onClick={this.deleteChannel}
                                >
                                    Delete this Channel
                                </Button>{" "}
                                {!!posts && posts.length > 0 && (
                                    <Button
                                        color="primary"
                                        variant="contained"
                                        className="button"
                                        onClick={this.deleteAllPosts}
                                    >
                                        Delete All Posts
                                    </Button>
                                )}
                            </div>
                        )}
                        <Box marginTop={2}>
                            <MyPostFilterSelector />
                        </Box>
                    </Container>
                </Box>
                {posts !== [] && (
                    <Box paddingTop={2}>
                        <Container fixed>
                            <div className="polaroid__container">
                                {posts.length > 0 ? (
                                    posts.map((post) => {
                                        return (
                                            <MyPostSummary
                                                post={post}
                                                key={post.post_id}
                                                pathName={`/myPosts/${post.post_id}/edit`}
                                                readOnly={this.props.channel.deleted_flag}
                                                inHorizontalMenu={false}
                                            />
                                        );
                                    })
                                ) : (
                                    <Typography color="error" variant="h4">
                                        No Channels
                                    </Typography>
                                )}
                            </div>
                        </Container>
                    </Box>
                )}
            </div>
        );
    }
}

const mapStateToProps = (state) => {
    return {
        filters: state.postFilters,
        channel: state.channels.channels.length === 1 && state.channels.channels[0],
        loading: state.channels.loading,
        posts:
            state.channels.channels.length === 1
                ? getVisiblePosts(
                      state.channels.channels[0].channel_posts,
                      state.postFilters,
                      state.channels.channels[0].deleted_flag
                  )
                : []
    };
};

const mapDispatchToProps = (dispatch) => ({
    startGetChannel: (id) => dispatch(startGetChannel(id)),
    deleteChannel: (id) => dispatch(deleteChannel(id)),
    restoreChannel: (id) => dispatch(restoreChannel(id)),
    deletePost: (id) => dispatch(deletePost(id))
});

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(MyChannelDetail);
