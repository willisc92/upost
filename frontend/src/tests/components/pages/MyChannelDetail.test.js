import { shallow } from "enzyme";
import { MyChannelDetail } from "../../../components/pages/MyChannelDetail";
import React from "react";
import channels from "../../fixtures/channels";
import posts from "../../fixtures/posts";

let startGetChannel, wrapper, history, match, channel;

beforeEach(() => {
    startGetChannel = jest.fn().mockImplementationOnce(() => {
        return Promise.resolve();
    });
    localStorage.setItem("user_name", channels[0].user);
    history = {
        push: jest.fn()
    };
    channel = channels[0];
    match = {
        params: {
            id: channel.channel_id
        }
    };
    wrapper = shallow(
        <MyChannelDetail
            startGetChannel={startGetChannel}
            match={match}
            loading={false}
            channel={channel}
            posts={posts}
            history={history}
        />
    );
});

test("Should render page correctly", () => {
    expect(wrapper).toMatchSnapshot();
});

test("Should handleAddPost", () => {
    wrapper
        .find("button")
        .at(0)
        .simulate("click");
    expect(history.push).toHaveBeenCalled();
    expect(history.push).toHaveBeenCalledWith(`/myChannels/${match.params.id}/addPost`);
});

test("Should handleEditChannel", () => {
    wrapper
        .find("button")
        .at(1)
        .simulate("click", {
            target: { id: channel.channel_id }
        });
    expect(history.push).toHaveBeenCalled();
    expect(history.push).toHaveBeenCalledWith(`/myChannels/edit/${channel.channel_id}`);
});
