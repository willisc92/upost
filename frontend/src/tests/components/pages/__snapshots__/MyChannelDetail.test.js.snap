// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render page correctly 1`] = `
<div>
  <div
    className="page-header"
  >
    <div
      className="content-container"
    >
      <h1
        className="page-header__title"
      >
        Channel Page
      </h1>
      <div>
        <h2>
          Name: 
          Channel 1
        </h2>
        <h3>
          Description: 
          Test
        </h3>
        <h3>
          Creation Date: 
          June 4th 2019
        </h3>
        <div>
          <button
            className="button button--secondary"
            onClick={[Function]}
          >
            Add a new post
          </button>
          <span>
             
          </span>
          <button
            className="button button--secondary"
            id={1}
            onClick={[Function]}
          >
            Edit this channel
          </button>
        </div>
        <div
          className="page-header__actions"
        >
          <Connect(ChannelFilters) />
        </div>
      </div>
    </div>
  </div>
  <div
    className="content-container"
  >
    <div
      className="list-body"
    >
      <div
        className="list-parent"
      >
        <p>
           No posts 
        </p>
      </div>
    </div>
  </div>
</div>
`;
